src/defines.h

```diff
@@ -15,6 +15,19 @@
 
 #define	DEBUG					0			// Set to 0 for production, 1 for extra debug information
 
+// ZDS segment stuff
+extern void _heapbot[];
+extern void _heaptop[];
+extern void _stack[];
+extern void _low_data[];
+extern void _low_bss[];
+extern void _low_romdata[];
+extern int _len_data;
+
+
+#define SPL_STACK_SIZE				1024
```

Comment:
I'm curious as to whether you've tried smaller stack sizes?  could we get away with just 512 bytes, and thus get 512 bytes more heap?

1kb seems reasonable tho

Reply:
experiments showed that 1024 bytes of stack doesn't seem to be quite enough.  a stack of 1.5kb appears to be more reliable